lib/mods_display/fields/date_other.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module ModsDisplay
+  class DateOther < Field
+    def fields
+      date_fields(:dateOther)
+    end
+
+    private
+
+    # overriding the original method in Field in order to include the type attribute value in parens
+    def select_best_date(dates)
+      # ensure dates are unique with respect to their base values
+      dates = dates.group_by(&:base_value).map do |_value, group|
+        group.first if group.one?
+
+        # if one of the duplicates wasn't encoded, use that one. see:
+        # https://consul.stanford.edu/display/chimera/MODS+display+rules#MODSdisplayrules-3b.%3CoriginInfo%3E
+        if group.reject(&:encoding).any?
+          group.reject(&:encoding).first
+
+        # otherwise just randomly pick the last in the group
+        else
+          group.last
+        end
+      end
+
+      # if any single dates are already part of a range, discard them
+      range_base_values = dates.select { |date| date.is_a?(Stanford::Mods::Imprint::DateRange) }
+                               .map(&:base_values).flatten
+      dates = dates.reject { |date| range_base_values.include?(date.base_value) }
+
+      # output formatted dates with qualifiers, CE/BCE, etc.
```

Comment:
Rather than duplicate the entire method, can this method be extracted into a class that allows just the relevant part to be overwritten?

Reply:
Take a look at how I implemented this now and see if you're ok with it.   I think it's clearer now than it would be with a separate class.